federator/src/lib/FederatorHTR.ts

```diff
@@ -134,7 +134,7 @@ export default class FederatorHTR extends Federator {
     this.logger.upsertContext('Current Block', getLogParams.currentBlock);
     const mainBridge = await getLogParams.bridgeFactory.createInstance(this.config.mainchain);
 
-    const recordsPerPage = 500;
+    const recordsPerPage = 450;
     const numberOfPages = Math.ceil((getLogParams.toBlock - getLogParams.fromBlock) / recordsPerPage);
```

Comment:
I think the real issue causing some of the federators to request more than 500 blocks in a page lies in this line.

I'll give an example. Suppose `toBlock = 10` and `fromBlock = 1`, and `recordsPerPage = 3`.

The result of `numberOfPages` would be 10 - 1 / 3 = 3.
So it would actually try to fit 10 blocks into 3 pages, resulting in pages of size 3, 3 and 4 (because you don't limit the size of the last page, line 145)

The correct calculation of the number of pages should be `const numberOfPages = Math.ceil((getLogParams.toBlock - getLogParams.fromBlock + 1) / recordsPerPage);`, to make sure the last page size is not bigger than `recordsPerPage`.

I think this would solve the problem even if we keep `recordsPerPage = 500`.